version: '3.9'
services:
  mysql:
    image: mysql:5.7.38
    container_name: mysql
    environment:
      MYSQL_ROOT_PASSWORD: ''
      MYSQL_ALLOW_EMPTY_PASSWORD: 'yes'
      TZ: 'Asia/Tokyo'
    ports:
      - 3306:3306
    volumes:
      - ./data/mysql:/var/lib/mysql
      - ./mysql/sql:/docker-entrypoint-initdb.d
      - ./mysql/my.cnf:/etc/mysql/conf.d/my.cnf
  # アプリケーション側では使っていない（結合テストを試してみたかった時に使っていた）
  # dynamodb-local:
  #   command: -jar DynamoDBLocal.jar -sharedDb -dbPath ./data
  #   image: 'amazon/dynamodb-local:latest'
  #   container_name: dynamodb-local
  #   ports:
  #     - '8000:8000'
  #   volumes:
  #     - './data/dynamodb:/home/dynamodblocal/data'
  #   working_dir: /home/dynamodblocal
  #   user: root
  # アプリケーション側では使っていない（Redisのクライアントioredisのコネクション確認で使っていた）
  # redis:
  #   image: redis:6.2.6-alpine3.15
  #   container_name: redis
  #   environment:
  #     TZ: 'Asia/Tokyo'
  #   volumes:
  #     - ./data/redis:/data
  #   ports:
  #     - 6379:6379
